src/datamodel/validator/UMessage.cpp

```diff
@@ -50,34 +47,37 @@ std::string_view message(Reason reason) {
 		case Reason::URI_MISMATCH:
 			return "The source/sink URIs for the request/response pair have "
 			       "not been swapped";
+		case Reason::WRONG_MESSAGE_TYPE:
+			return "The type set in the message is incorrect for the validated "
+			       "mode";
 		default:
 			return "Unknown reason.";
 	}
 }
 
 ValidationResult isValid(const v1::UMessage& umessage) {
```

Comment:
I wonder if we'd have a better "isValid" with better error codes if we respect the umessage.attributes.type() field ... like this:

`        if (umessage.attributes().type() ==
            v1::UMessageType::UMESSAGE_TYPE_REQUEST) {
                return isValidRpcRequest(message);
        }
`

That way if a message says "I'm a request" you only test for that?  Since every other type will fail with WRONG_MESSAGE_TYPE.

Reply:
That's not a bad idea :thinking: 

It would also be more efficient. I'll give it a try

Reply:
Done